src/Nox.Solution/Models/Domain/Entities/Relationships/EntityRelationship.cs

```diff
@@ -41,6 +41,15 @@ public class EntityRelationship : DefinitionBase
     [YamlIgnore]
     public bool ShouldGenerateForeignOnThisSide => NoxSolution.ShouldGenerateForeignOnThisSide(this);
 
+    [YamlIgnore]
+    public bool HasRelationshipWithSingularEntity => NoxSolution.HasRelationshipWithSingularEntity(this);
```

Comment:
this circular dependency should be avoid, all this 3 methods (HasRelationshipWithSingularEntity, .ShouldUseRelationshipNameAsNavigation,IsManyRelationshipOnOtherSide) should be part of the EntityRelationship class

That is the only info they need